workers/wfsfeature-harvester/src/main/java/org/fao/geonet/harvester/wfsfeatures/worker/WFSClientWithStrategyInvestigator.java

```diff
@@ -84,12 +86,16 @@ private WFSStrategy determineCorrectStrategy(HTTPClient httpClient) {
             while ((readCount = inputStream.read(buff)) != -1) {
                 out.write(buff, 0, readCount);
             }
-            String responsePayload = out.toString("UTF-8");
+            String responsePayload = out.toString(StandardCharsets.UTF_8);
             if (responsePayload.contains("targetNamespace=\"http://www.qgis.org/gml\"")) {
                 strategy = new QgisStrategy();
                 strategyId = QGIS_STRATEGY;
             } else if (responsePayload.contains("targetNamespace=\"http://mapserver.gis.umn.edu/mapserver\"")) {
-                strategy = new MapServerWFSStrategy(capabilities.getRawDocument());
+                if (capabilities.getVersion().equals("2.0.0")) {
```

Comment:
Maybe add  the value `2.0.0` or check if Geotools has a constant already defined for WFS 2.0.0